misk/src/test/kotlin/misk/tracing/interceptors/TracingMethodInterceptorTest.kt

```diff
@@ -0,0 +1,42 @@
+package misk.tracing.interceptors
+
+import com.google.inject.Guice
+import com.google.inject.util.Modules
+import io.opentracing.Tracer
+import io.opentracing.mock.MockTracer
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.testing.MockTracingBackendModule
+import misk.tracing.Trace
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest
+class TracingMethodInterceptorTest {
+  @MiskTestModule
+  val module = Modules.combine(
+      TracingMethodInterceptorModule(),
+      MockTracingBackendModule()
+  )
+
+  @Inject private lateinit var tracer: Tracer
+
+  @Inject private lateinit var testObject: TestObject
+
+  @Test
+  fun traceAnnotationTraces() {
+    val mockTracer = tracer as MockTracer
+
+    assertThat(mockTracer.finishedSpans().size).isEqualTo(0)
+    testObject.testTraceMethod()
+    assertThat(mockTracer.finishedSpans().size).isEqualTo(1)
+  }
+}
+
+open class TestObject {
+  @Trace
+  open fun testTraceMethod() {
```

Comment:
so what did you need to do to get this working?

Reply:
Actually mark the function open in addition to the class 🤦‍♂️